floodlight/io/catapultApi.py

```diff
@@ -0,0 +1,384 @@
+    t_null = timestamps_cs[0]
+
+    return pID_dict, number_of_frames, framerate, t_null
+
+def _get_available_sensor_identifier(pID_dict: Dict[str, Dict[str, List[str]]]) -> str:
+    """
+    Determines an available sensor identifier from the metadata.
+
+    Args:
+        pID_dict (Dict[str, Dict[str, List[str]]]): Metadata dictionary containing player IDs by group.
+
+    Returns:
+        str: The available sensor identifier.
+    """
+    player_identifiers = ["name", "mapped_id", "sensor_id", "number"]
+    available_identifier = [idt for idt in player_identifiers if idt in list(pID_dict.values())[0]]
+    return available_identifier[0]
+
+def create_links_from_meta_data(
+    pID_dict: Dict[str, Dict[str, List[str]]], identifier: str = None
+) -> Dict[str, Dict[str, int]]:
+    """
+    Creates a mapping of player IDs to numerical indices based on the metadata.
+
+    Args:
+        pID_dict (Dict[str, Dict[str, List[str]]]): Metadata dictionary containing player IDs by group.
+        identifier (str, optional): The sensor identifier to use for mapping. Defaults to the available identifier.
+
+    Returns:
+        Dict[str, Dict[str, int]]: A dictionary mapping group IDs to player ID indices.
+    """
+    if identifier is None:
+        identifier = _get_available_sensor_identifier(pID_dict)
+
+    links = {}
+    for group_id in pID_dict:
+        links[group_id] = {ID: xID for xID, ID in enumerate(pID_dict[group_id][identifier])}
+
+    return links
+
+def read_position_data_from_json(response_data_json_list: List[Dict], coord_format: str = "xy") -> List[XY]:
+    """
+    Reads position data from JSON data responses and returns a list of XY objects.
+
+    Args:
+        response_data_json_list (List[Dict]): List of JSON data responses.
+        coord_format (str, optional): The coordinate format ["xy", "latlong"]. Defaults to "xy".
+
+    Returns:
+        List[XY]: A list of XY objects containing position data.
+    """
+    pID_dict, number_of_frames, framerate, t_null = get_meta_data(response_data_json_list)
+
+    links = create_links_from_meta_data(pID_dict)
+
+    key_links = _get_key_links(response_data_json_list)
+    key_links_set = set(key_links)
+    group_identifier_set = {"group_id", "group_name"}
+    recorded_group_identifier = list(key_links_set & group_identifier_set)
+    identifier = _get_available_sensor_identifier(pID_dict)
+
+    reversed_group_mapping = {
+        "group_id": "team_id",
+        "group_name": "team_name",
+    }
+    number_of_sensors = {}
+    xydata = {}
+
+    for group in links:
+        number_of_sensors.update({group: len(links[group])})
+        xydata.update({group: np.full([number_of_frames + 1, number_of_sensors[group] * 2], np.nan)})
+
+    for data_i in response_data_json_list:
+        for entry in data_i["data"]:
+            if coord_format == "xy":
+                x_coordinate = entry["x"] / 10  # convert mm to cm

```

Comment:
I think the Catapult documentation states that these values are in m. Nevertheless, we should not scale them in the parsing process up or down when the user does not expect it.

Reply:
Done